ui/tests/integration/components/ttl-picker2-test.js

```diff
@@ -0,0 +1,117 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, click, fillIn } from '@ember/test-helpers';
+import sinon from 'sinon';
+import hbs from 'htmlbars-inline-precompile';
+
+module('Integration | Component | ttl-picker2', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders time and unit inputs when TTL enabled', async function(assert) {
+    let changeSpy = sinon.spy();
+    this.set('onChange', changeSpy);
+    await render(hbs`
+      <TtlPicker2
+        @onChange={{onChange}}
+        @enableTTL={{true}}
+      />
+    `);
+
+    assert.dom('[data-test-ttl-value]').exists('TTL Picker time input exists');
+    assert.dom('[data-test-ttl-unit]').exists('TTL Picker unit select exists');
+  });
+
+  test('it does not show time and unit inputs when TTL disabled', async function(assert) {
+    let changeSpy = sinon.spy();
+    this.set('onChange', changeSpy);
+    await render(hbs`
+      <TtlPicker2
+        @onChange={{onChange}}
+        @enableTTL={{false}}
+      />
+    `);
+    assert.dom('[data-test-ttl-value]').doesNotExist('TTL Picker time input exists');
+    assert.dom('[data-test-ttl-unit]').doesNotExist('TTL Picker unit select exists');
+  });
+
+  test('it passes the appropriate data to onChange when toggled on', async function(assert) {
+    let changeSpy = sinon.spy();
+    this.set('onChange', changeSpy);
+    await render(hbs`
+      <TtlPicker2
+        @label="clicktest"
+        @unit="m"
+        @time="10"
+        @onChange={{onChange}}
+        @enableTTL={{false}}
+      />
+    `);
+    await click('[data-test-toggle-input="clicktest"]');
+    assert.ok(changeSpy.calledOnce, 'it calls the passed onChange');
+    assert.ok(
+      changeSpy.calledWith({
+        enabled: true,
+        seconds: 600,
+        timeString: '10m',
+      }),
+      'Passes the default values back to onChange'
```

Comment:
Yes, and also verifying that it uses the `time` and `unit` values that were passed in instead of its own defaults. I guess that's sort of just testing the parameter setting of Ember, but I can for sure rephrase the test so it reflects the action instead